docs(metrics): Add search bar documentation (#5394)
getsentry/sentry-docs: src/docs/product/performance/filters-display/index.mdx

"Mobile" puts an emphasis on [Mobile Vitals](/product/performance/mobile-vitals/). Note that if you've selected a non-mobile project, this mode also displays all of those transactions.

> ## Transaction Search

Reviewer: This section feels like it fits nicely under the heading just below! You could rename "Search Condition Filters" to "Transaction Search", and combine them. Both sections are about the search bar, and this `<Note>` could be a caveat

<Note>

This feature is in beta and is only available if your organization is participating in its limited release. Features in beta are still in-progress and may have bugs. We recognize the irony. If you’re interested in participating, [join the waitlist](https://sentry.io/for/performance/#updates-signup).

</Note>

> The transaction filter on the performance page has been simplified to support metric enhanced performance. The search bar can only be used to filter by transaction name.

Reviewer: Maybe helpful to clarify the difference from the regular search bar. Maybe something like "The search bar can only be used to filter by transaction name, rather than to search for transactions matching the query".

Reviewer: Also, I wonder if MEP is a marketing-visible feature, in which case it should be capitalized?

Author: As suggested by @nanoburd we replaced the MEP with metrics, as metrics has documentation to link to. Also, I agree, and reworded the search bar docs as suggested!

![Updated transaction search](transaction-name-search.png)

> When you begin typing a list of transaction names that match your search appear. There are two ways to interact with the search results.

Reviewer 2: ```suggestion
When you begin typing, a list of transaction names that match your search appear. There are two ways to interact with the search results:
```

> 1. Clicking on a transaction name will apply a filter on the performance page to only display transaction data for the selected transaction.

Reviewer 2: ```suggestion
1. Clicking on a transaction name will apply a filter on the Performance page to only display transaction data for the selected transaction.
```

> 2. Clicking the upwards arrow on a search result will take you to that transactions summary page.

Reviewer 2: ```suggestion
2. Clicking the upwards arrow icon on a search result will take you to that transaction's summary page.
```
